src/components/ImageView/index.native.js

```diff
@@ -86,14 +57,56 @@ class ImageView extends PureComponent {
      * @param {GestureState} gestureState
      * @returns {Boolean}
      */
-    updatePanResponderTouches(e, gestureState) {
+    const updatePanResponderTouches = (e, gestureState) => {
         if (_.isNumber(gestureState.numberActiveTouches)) {
-            this.amountOfTouches = gestureState.numberActiveTouches;
+            numberOfTouches.current = gestureState.numberActiveTouches;
         }
 
         // We don't need to set the panResponder since all we care about is checking the gestureState, so return false
         return false;
-    }
+    };
+
+    // PanResponder used to capture how many touches are active on the attachment image
+    const panResponder = React.useRef(
+        PanResponder.create({
+            onStartShouldSetPanResponder: updatePanResponderTouches,
+        }),
+    ).current;
+
+    /**
+     * When the url changes and the image must load again,
+     * this resets the zoom to ensure the next image loads with the correct dimensions.
+     */
+    const resetImageZoom = React.useCallback(() => {
+        if (imageZoomScale.current !== 1) {
+            imageZoomScale.current = 1;
+        }
+
+        if (zoom.current) {
+            zoom.current.centerOn({
+                x: 0,
+                y: 0,
+                scale: 1,
+                duration: 0,
+            });
+        }
+    }, []);
+
+    const imageLoadingStart = () => {
+        if (isLoading) {
+            return;
+        }
+        resetImageZoom();
+
+        setImageHeight(0);
+        setImageWidth(0);
+        setIsLoading(true);
+    };
+
+    React.useEffect(() => {
+        imageLoadingStart();
+        // eslint-disable-next-line react-hooks/exhaustive-deps -- adding imageLoadingStart to deps array causes the component to be stuck on loading
+    }, [url]);
```

Comment:
did you try to create the `imageLoadingStart` with a `useCallback`, maybe because it isn't a stable function and on a rerender the `useEffect` gets called again. Try using the `useCallback` first instead of remove it from the dependency array, if not please a more detailed explanation why not adding it

Reply:
I have tried, and the component failed to load the image, the effect was called too many times and therefore it got stuck on loading.

Reply:
I added a more detailed explanation as to why we need to disable this rule. Let me know if that's enough!